sql/src/main/java/io/crate/planner/node/ddl/RestoreSnapshotPlan.java

```diff
@@ -0,0 +1,322 @@
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.node.ddl;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.crate.analyze.AnalyzedRestoreSnapshot;
+import io.crate.analyze.BoundedRestoreSnapshot;
+import io.crate.analyze.GenericPropertiesConverter;
+import io.crate.analyze.SnapshotSettings;
+import io.crate.analyze.SymbolEvaluator;
+import io.crate.common.collections.Lists2;
+import io.crate.data.Row;
+import io.crate.data.Row1;
+import io.crate.data.RowConsumer;
+import io.crate.exceptions.PartitionAlreadyExistsException;
+import io.crate.exceptions.RelationAlreadyExists;
+import io.crate.exceptions.RelationUnknown;
+import io.crate.execution.support.OneRowActionListener;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.Functions;
+import io.crate.metadata.IndexParts;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.Operation;
+import io.crate.planner.DependencyCarrier;
+import io.crate.planner.Plan;
+import io.crate.planner.PlannerContext;
+import io.crate.planner.operators.SubQueryResults;
+import io.crate.sql.tree.Table;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsRequest;
+import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsResponse;
+import org.elasticsearch.action.admin.cluster.snapshots.get.TransportGetSnapshotsAction;
+import org.elasticsearch.action.admin.cluster.snapshots.restore.RestoreSnapshotRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.snapshots.SnapshotInfo;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static io.crate.analyze.PartitionPropertiesAnalyzer.toPartitionName;
+import static io.crate.analyze.SnapshotSettings.IGNORE_UNAVAILABLE;
+import static io.crate.analyze.SnapshotSettings.WAIT_FOR_COMPLETION;
+
+public class RestoreSnapshotPlan implements Plan {
+
+    private final String[] ALL_TEMPLATES = new String[]{"_all"};
```

Comment:
Minor, `_ALL_TEMPLATES` is only used once, so there is not really a need to keep the array permanently in memory, but just create it once it is needed, but i am also ok with the way it is.
```suggestion
    private final String ALL_TEMPLATES = "_all";
```